include/data_struct.h

```diff
@@ -197,6 +217,13 @@ class BaseLSTMStates {
     void reset_zeros();
 };
 
+// Smoother for linear layer when using LSTM
+class BaseSLinear {
```

Comment:
This class is so confusing! I dont think we need to define a class here for a particular case